runtime/runtime_controller.h

```diff
@@ -41,53 +117,329 @@ class RuntimeController final : public WindowClient {
+  ///             new frame.
+  ///
+  /// @see        `Engine::BeginFrame` for more context.
+  ///
+  /// @param[in]  frame_time  The point at which the current frame interval
+  ///                         began. May be used by animation interpolators,
+  ///                         physics simulations, etc..
+  ///
+  /// @return     If notification to begin frame rendering was delivered to the
+  ///             running isolate.
+  ///
   bool BeginFrame(fml::TimePoint frame_time);
 
+  //----------------------------------------------------------------------------
+  /// @brief      Dart code cannot fully measure the time it takes for a
+  ///             specific frame to be rendered. This is because Dart code only
+  ///             runs on the UI task runner. That is only a small part of the
+  ///             overall frame workload. The GPU task runner frame workload is
+  ///             executed on a thread where Dart code cannot run (and hence
+  ///             instrument). Besides, due to the pipelined nature of rendering
+  ///             in Flutter, there may be multiple frame workloads being
+  ///             processed at any given time. However, for non-Timeline based
+  ///             profiling, it is useful for trace collection and processing to
+  ///             happen in Dart. To do this, the GPU task runner frame
+  ///             workloads need to be instrumented separately. After a set
+  ///             number of these profiles have been gathered, they need to be
+  ///             reported back to Dart code. The engine reports this extra
+  ///             instrumentation information back to Dart code running on the
+  ///             engine by invoking this method at predefined intervals.
+  ///
+  /// @see        `Engine::ReportTimings`, `FrameTiming`
+  ///
+  /// @param[in]  timings  Collection of `FrameTiming::kCount` * `n` timestamps
+  ///                      for `n` frames whose timings have not been reported
+  ///                      yet. A collection of integers is reported here for
+  ///                      easier conversions to Dart objects. The timestamps
+  ///                      are measured against the system monotonic clock
+  ///                      measured in microseconds.
+  ///
   bool ReportTimings(std::vector<int64_t> timings);
 
+  //----------------------------------------------------------------------------
+  /// @brief      Notify the Dart VM that no frame workloads are expected on the
+  ///             UI task runner till the specified deadline. The VM uses this
+  ///             opportunity to perform garbage collection operations is a
+  ///             manner that interferes as little as possible with frame
+  ///             rendering.
+  ///
+  /// NotifyIdle is advisory and not making the same or making it with
+  /// insufficient deadlines does not mean that the VM will keep consuming
+  /// memory indefinitely. Even if the engine made absolutely no calls to
```

Comment:
This would be a little clearer if phrased without the double negative... something like

>NotifyIdle is advisory. The VM may or may not run a garbage collection when this is called, and will eventually perform garbage collections even if it is not called or it is called with insufficient deadlines.

?

Reply:
Patching.